src/scripts/axis/create.js

```diff
@@ -37,7 +37,7 @@ function timeScale(model, options, orientation) {
 
 function linearScale(model, options, orientation) {
     var range = getRange(model, orientation);
-    var domain = (isVertical(orientation)) ? model.dependentDomain.reverse() : model.dependentDomain;
+    var domain = (isVertical(orientation) && !model.y.reverse) ? model.dependentDomain.reverse() : model.dependentDomain;
```

Comment:
This makes sense but i'd favour an if-else style, for clarity:

```js
var domain = model.dependentDomain;
if (isVertical(orientation) && !model.y.reverse) { 
  model.dependentDomain.reverse()
}
```

I'm not a big fan of inline ifs unless you're inside objects.